### ISSUE TITLE
/downloads not working on staging -  cannot convert to or from a Postgres value of type `date`

### ISSUE THREAD

**Issue by carols10cents** [rust-lang/crates.io#648]
I merged https://github.com/rust-lang/crates.io/pull/641, deployed to staging, and THEN noticed that there's an error when you request downloads on a crate page :( At least I caught it before it went to prod?

I can reproduce this locally too. When I visit a crate's page, the request to /downloads for that crate errors and I see this in the server output:

```
Get /api/v1/crates/piston/downloads - 11ms 500: Error converting between Postgres and Rust types Error converting between Postgres and Rust types: cannot convert to or from a Postgres value of type `date`
```

Not sure if it's the database or the rust code that's wrong.... can you fix this @sgrif ?

**Comment by carols10cents** [comment]
ughhh i don't understand why [this test](https://github.com/rust-lang/crates.io/blob/master/src/tests/krate.rs#L759) didn't fail

**Comment by carols10cents** [comment]
Ok, i give up. I can't figure out what's wrong, it's something with rust postgres, so perhaps moving /downloads over to use diesel too is the way forward?

I've reverted PR #641 until we can figure this out. I reverted the  20170322130523_version_downloads_needs_no_id migration on staging, the  20170318181441_downloads_need_no_id one wouldn't revert:

```
Rolling back migration 20170318181441
syntax error at or near "PRIMARY"
```

 but doesn't seem to be a problem.

Prod is fine because I never deployed #641 there so I don't have to roll back migrations since they were never made.
